src/common/simulation.ts

```diff
@@ -213,28 +263,48 @@ export default abstract class Simulation {
       body.getPosition(),
       Vec2.mul(direction, multiplier),
     );
+    let closestTarget: {
+      fixture?: Fixture;
+      fraction: number;
+      point?: Vec2;
+    } = {
+      fraction: Infinity,
+    };
+
+    this.world.rayCast(
+      body.getPosition(),
+      endPoint,
+      (fixture, point, normal, fraction) => {
+        if (fraction < closestTarget.fraction) {
+          closestTarget = { fixture, fraction, point };
+        }
+        return fraction;
+      },
+    );
 
-    this.world.rayCast(body.getPosition(), endPoint, (...args) =>
-      this.rayCastCallback(...args, player),
+    this.handleHit(
+      closestTarget.fixture!,
+      closestTarget.point!,
```

Comment:
```suggestion
      closestTarget.fixture,
      closestTarget.point,
```
Ser ut som att min commit har hämnat här, fixa detta